src/en/darthsdroids/src/eu/kanade/tachiyomi/extension/en/darthsdroids/DarthsDroids.kt

```diff
@@ -0,0 +1,222 @@
+    // whether we’re in a date-annotated archive, the main archive, or a dateless archive.
+    // All three are largely similar, there are just *some* (annoying) differences we have to
+    // deal with.
+    private fun fetchChaptersForDatedAndDatelessArchives(archivePages: Document, archivePath: String): List<SChapter> {
+        val chapters = mutableListOf<SChapter>()
+        var i = 0
+        archivePages.select("""div.text > table.text > tbody > tr""").forEach {
+            val pageData = it.select("""td""")
+            var pageAnchor = pageData.getOrNull(2)?.selectFirst("a")
+            // null for »Intermission«, main archive, dateless archive,…
+            if (pageAnchor != null) {
+                val a = pageAnchor
+
+                chapters.add(
+                    SChapter.create().apply {
+                        name = a.text()
+                        chapter_number = i.toFloat()
+                        date_upload = DATE_FMT.parse(pageData[0].text().trim())?.time ?: 0L
+                        url = absolutePageUrl(a.attr("href"), archivePath)
+                    },
+                )
+                i += 1
+            } else if (!pageData.hasAttr("colspan")) {
+                // Are we in a dateless archive?
+                pageAnchor = pageData.getOrNull(0)?.selectFirst("a")
+                if (pageAnchor != null) {
+                    // For now we assume all dateless archives were published the same day
+                    // the Solo book was published, because currently Solo’s the only one such book.
+                    val pageDate = 1660435200000L + (i * 1000 * 60)
+                    val a = pageAnchor
+
+                    chapters.add(
+                        SChapter.create().apply {
+                            name = a.text()
+                            chapter_number = i.toFloat()
+                            date_upload = pageDate
+                            url = absolutePageUrl(a.attr("href"), archivePath)
+                        },
+                    )
+                    i += 1
+                }
+            }
+        }
+        chapters.reverse()
+        return chapters
+    }
+
+    private fun absolutePageUrl(pageUrl: String, maybeRelativeTo: String): String =
+        if (pageUrl.startsWith('/')) {
+            pageUrl
+        } else {
+            // This happens because for whatever reason, the `solo` index isn’t
+            // a `.html` page, but a directory path with an `index.html`. So
+            // we’re returning `solo/$pageUrl`.
+            maybeRelativeTo + pageUrl
+        }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        var archiveResponse = client.newCall(GET(baseUrl + manga.url, headers)).execute()
+        if (!archiveResponse.isSuccessful) {
+            archiveResponse = client.newCall(GET(mainArchiveUrl, headers)).execute()
```

Comment:
I forgot why exactly i made it that way, but it may have something to do with the `/archive.html` eventually changing. Will investigate a better way. 

I *think* i did it that way because at one point i had the code special-case all comic book initialisation of `SManga`, and book 8 preëmptively had its archive page set to `/archive8.html`. I.e. it’s legacy code.

Reply:
Method nuked, only `chapterListParse` now remains.